v3/src/components/graph/components/barchartdots.tsx

```diff
@@ -0,0 +1,184 @@
+import { ScaleBand, ScaleLinear, select } from "d3"
+import React, { useCallback, useEffect, useRef } from "react"
+import { createPortal } from "react-dom"
+import { CellType, IBarCover, PlotProps } from "../graphing-types"
+import { usePlotResponders } from "../hooks/use-plot"
+import { handleClickOnBar } from "../../data-display/data-display-utils"
+import { setPointCoordinates } from "../utilities/graph-utils"
+import { IDataConfigurationModel } from "../../data-display/models/data-configuration-model"
+import { GraphLayout } from "../models/graph-layout"
+import { AxisPlace } from "../../axis/axis-types"
+import { mstAutorun } from "../../../utilities/mst-autorun"
+import { useChartDots } from "../hooks/use-chart-dots"
+
+interface IRenderBarCoverProps {
+  barCovers: IBarCover[]
+  barCoversRef: React.RefObject<SVGGElement>
+  dataConfiguration: IDataConfigurationModel
+  primaryAttrRole: "x" | "y"
+}
+
+interface IBarCoverDimensionsProps {
+  cellIndices: CellType
+  layout: GraphLayout
+  extraPrimaryAxisPlace: AxisPlace
+  maxInCell: number
+  numExtraSecondaryBands: number
+  primaryAxisPlace: AxisPlace
+  primaryIsBottom: boolean
+  primCatsCount: number
+  secondaryAxisPlace: AxisPlace
+  secondaryCellHeight: number
+}
+
+const renderBarCovers = (props: IRenderBarCoverProps) => {
+  const { barCovers, barCoversRef, dataConfiguration, primaryAttrRole } = props
+  select(barCoversRef.current).selectAll("rect").remove()
+  select(barCoversRef.current).selectAll("rect")
+    .data(barCovers)
+    .join((enter) => enter.append("rect")
+      .attr("class", (d) => d.class)
+      .attr("data-testid", "bar-cover")
+      .attr("x", (d) => d.x)
+      .attr("y", (d) => d.y)
+      .attr("width", (d) => d.width)
+      .attr("height", (d) => d.height)
+      .on("mouseover", function() { select(this).classed("active", true) })
+      .on("mouseout", function() { select(this).classed("active", false) })
```

Comment:
For some reason this isn't causing the datatip to disappear.

Reply:
I meant to change this but didn't: currently the individual PixiPoint bars are still handling the showing and hiding of data tips. It should be the bar covers that do so. I'll need to rework things a little. Talya also pointed out that the click handling wasn't working right when there's a legend present and the bars are segmented by legend values. So I'll also have to make it so there's a bar cover per bar segment when there's a legend.

Reply:
As noted offline, there is now a [separate PT story for this bug](https://www.pivotaltracker.com/story/show/187409553).